x/masterchef/keeper/abci.go

```diff
@@ -747,7 +747,8 @@ func (k Keeper) ProcessTakerFee(ctx sdk.Context) {
 		if balance.Denom == ptypes.Elys || balance.Amount.LT(sdkmath.NewInt(1000000)) {
 			continue
 		}
-		_, err := k.amm.SwapByDenom(ctx, &ammtypes.MsgSwapByDenom{
+		cacheCtx, write := ctx.CacheContext()
```

Comment:
SHall we extennd this cacheCtx to all conversions in masterchef abci, if we start taking fees in some 18 decimal token and prices drops by a lot then, conversion would fail leading the abci to write incorrect state